tests/integration/test_cli.py

```diff
@@ -41,52 +48,131 @@
 
 
 @pytest.mark.parametrize(
-    "bad_url,bad_outdir,file_id,max_wait_time,expected_exception",
+    "file_size,part_size",
     [
-        (True, False, "downloadable", "60", ApiNotReachable),
-        (False, False, "downloadable", "60", None),
+        (6 * 1024 * 1024, 5 * 1024 * 1024),
+        (12 * 1024 * 1024, 5 * 1024 * 1024),
+        (6 * 1024 * 1024, DEFAULT_PART_SIZE),
+        (20 * 1024 * 1024, DEFAULT_PART_SIZE),
+    ],
+)
+def test_multipart_download(
+    file_size,
+    part_size,
+    s3_fixture,  # noqa F811
+    tmp_path,
+):
+    """Test the multipart download of a file"""
+
+    with big_temp_file(file_size) as big_file:
+
+        object_fixture = ObjectFixture(
+            file_path=big_file.name, bucket_id="outbox", object_id="big-downloadable"
+        )
+
+        # upload file to s3
+        if not s3_fixture.storage.does_bucket_exist(object_fixture.bucket_id):
+            s3_fixture.storage.create_bucket(object_fixture.bucket_id)
+
+        if not s3_fixture.storage.does_object_exist(
+            bucket_id=object_fixture.bucket_id, object_id=object_fixture.object_id
+        ):
+            presigned_post = s3_fixture.storage.get_object_upload_url(
+                bucket_id=object_fixture.bucket_id, object_id=object_fixture.object_id
+            )
+            upload_file(
+                presigned_url=presigned_post,
+                file_path=big_file.name,
+                file_md5=object_fixture.md5,
+            )
```

Comment:
The s3_fixture already has existing objects for convenience . If you use them, you don't need this part.

Please have a look here: https://github.com/ghga-de/ghga-service-chassis-lib/blob/main/ghga_service_chassis_lib/s3_testing.py#L65
And here for an example: https://github.com/ghga-de/ghga-service-chassis-lib/blob/main/tests/integration/test_s3.py#L106

Reply:
The s3 fixture, however, has no big files available and a don't see a convenient way to inject them into the fixture.